l10n_it_fatturapa_out/models/account.py

```diff
@@ -95,11 +100,221 @@ def preventive_checks(self):
+            original_xml.findall(".//DatiBeniServizi/DettaglioLinee")
+        ):
+            raise UserError(
+                _(
+                    "{} isn't equal to tag in file e-invoice already created!".format(
+                        "DatiBeniServizi/DettaglioLinee"
+                    )
+                )
+            )
+        ld = 0
+        for new_line_details in new_xml.findall(".//DatiBeniServizi/DettaglioLinee"):
+            original_line_details = original_xml.findall(
+                ".//DatiBeniServizi/DettaglioLinee"
+            )[ld]
+            list_tag_DettaglioLinee = [
+                ".//NumeroLinea",
+                ".//CodiceTipo",
+                ".//CodiceValore",
+                ".//Descrizione",
+                ".//Natura",
+                ".//Ritenuta",
+            ]
+            self.check_tag(
+                new_line_details, original_line_details, list_tag_DettaglioLinee
+            )
+            list_tag_DettaglioLinee = [
+                ".//Quantita",
+            ]
+            self.check_tag(
+                new_line_details,
+                original_line_details,
+                list_tag_DettaglioLinee,
+                uom_precision,
+            )
+            list_tag_DettaglioLinee = [
+                ".//PrezzoUnitario",
+                ".//AliquotaIVA",
+                ".//PrezzoTotale",
+            ]
+            self.check_tag(
+                new_line_details,
+                original_line_details,
+                list_tag_DettaglioLinee,
+                price_precision,
+            )
+            ld += 1
+
+        if len(new_xml.findall(".//DatiBeniServizi/DatiRiepilogo")) != len(
+            original_xml.findall(".//DatiBeniServizi/DatiRiepilogo")
+        ):
+            raise UserError(
+                _(
+                    "{} isn't equal to tag in file e-invoice already created!".format(
+                        "DatiBeniServizi/DatiRiepilogo"
+                    )
+                )
+            )
+        lr = 0
+        for new_line_riepilogo in new_xml.findall(".//DatiBeniServizi/DatiRiepilogo"):
+            original_line_riepilogo = original_xml.findall(
+                ".//DatiBeniServizi/DatiRiepilogo"
+            )[lr]
+            list_tag_DatiRiepilogo = [
+                ".//AliquotaIVA",
+                ".//ImponibileImporto",
+                ".//Imposta",
+            ]
+            self.check_tag(
+                new_line_riepilogo,
+                original_line_riepilogo,
+                list_tag_DatiRiepilogo,
+                price_precision,
+            )
+            lr += 1
+
+    def elements_equal(self, new_xml, original_xml):
+        self.check_CessionarioCommittente(new_xml, original_xml)
+        self.check_DatiGeneraliDocumento(new_xml, original_xml)
+        self.check_DatiBeniServizi(new_xml, original_xml)
+
+    def write(self, vals):
+        is_draft = {}
+        for move in self:
```

Comment:
C'è una ragione particolare per cui crei questo dizionario invece di controllare direttamente lo `state` della move?

Reply:
sì, in pratica è per controllare qual è lo stato iniziale della move
io metto in bozza la fattura dopo la write la fattura già è in draft quendi poi entrerebbe nella funzione subito e si auto conferma

Reply:
ok chiaro, grazie.